fe/fe-core/src/main/java/org/apache/doris/load/loadv2/IngestionLoadJob.java

```diff
@@ -0,0 +1,1138 @@
+import org.apache.doris.thrift.TPriority;
+import org.apache.doris.thrift.TPushType;
+import org.apache.doris.thrift.TUniqueId;
+import org.apache.doris.transaction.BeginTransactionException;
+import org.apache.doris.transaction.TabletCommitInfo;
+import org.apache.doris.transaction.TabletQuorumFailedException;
+import org.apache.doris.transaction.TransactionState;
+
+import cfjd.com.google.gson.annotations.SerializedName;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Range;
+import com.google.common.collect.Sets;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import lombok.Setter;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.DataInput;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Ingestion Load
+ * </p>
+ * Load data file which has been pre-processed
+ * </p>
+ * There are 4 steps in IngestionLoadJob:
+ * Step1: Outside system execute ingestion etl job.
+ * Step2: LoadEtlChecker will check ingestion etl job status periodically
+ * and send push tasks to be when ingestion etl job is finished.
+ * Step3: LoadLoadingChecker will check loading status periodically and commit transaction when push tasks are finished.
+ * Step4: PublishVersionDaemon will send publish version tasks to be and finish transaction.
+ */
+public class IngestionLoadJob extends LoadJob {
+
+    public static final Logger LOG = LogManager.getLogger(IngestionLoadJob.class);
+
+    private long etlStartTimestamp = -1;
+
+    private long quorumFinishTimestamp = -1;
+
+    private List<Long> loadTableIds = new ArrayList<>();
+
+    @Setter
+    @SerializedName("ests")
+    private EtlStatus etlStatus;
+
+    private final Map<Long, Set<Long>> tableToLoadPartitions = Maps.newHashMap();
+
+    private final Map<Long, Map<Long, PushTask>> tabletToSentReplicaPushTask = Maps.newHashMap();
+
+    // members below updated when job state changed to loading
+    // { tableId.partitionId.indexId.bucket.schemaHash -> (etlFilePath, etlFileSize) }
+    @SerializedName(value = "tm2fi")
+    private final Map<String, Pair<String, Long>> tabletMetaToFileInfo = Maps.newHashMap();
+
+    private final Map<Long, Integer> indexToSchemaHash = Maps.newHashMap();
+
+    private final Map<String, Long> filePathToSize = new HashMap<>();
+
+    private final Set<Long> finishedReplicas = Sets.newHashSet();
+    private final Set<Long> quorumTablets = Sets.newHashSet();
+    private final Set<Long> fullTablets = Sets.newHashSet();
+
+    private final List<TabletCommitInfo> commitInfos = Lists.newArrayList();
+
+    @SerializedName(value = "hp")
```

Comment:
Move all fields with `@SerializedName` together

Reply:
Is it necessary to persist all fields? Not all fields need to be reloaded from metadata.

Reply:
done